### PR TITLE
Edit existing task

### LINKED ISSUE #4: Permettre l'edition d'une tache
Lorsque nous sommes sur la vue d'une tache, il faudrais que nous puissions editer cette tache avec la meme vue que la creation mais deja prepopuler des valeurs actuel.

Un bouton `...` est deja present en haut a droite de l'ecran.

### PR DESCRIPTION
**Pull request by kristofbc**
| Question               | Answer
| ---------------------- | -
| Bugs correction ?      | no
| New features ?         | yes
| Related issues         | closes #4  

---

##### What have you changed  ?
The TaskService with a new method: editTask (calling PUT: /task/:id);
The ModalAddTask handling the edit case

##### How did you do that ?
By adding a new method in TaskService and checking if there's a task in ModalAddTask: forcing it to edit instead of creating

### CODE REVIEW

File: tasklist/src/pages/modalAddTask/modalAddTask.ts [hunk 1]
```diff
@@ -83,21 +89,43 @@ export class ModalAddTask {
...
+
   dismiss() {
     this.viewCtrl.dismiss();
   }
 
   createForm() {
-    console.log(this.task);
-    this.taskService.createTask(this.task, this.project).subscribe(
+    let handler;
+    if( this.task.id ) {
```
Comment (RignonNoel) [inline tasklist/src/pages/modalAddTask/modalAddTask.ts]: I have an error when i launch the app :
```
[13:41:14]  typescript: src/pages/modalAddTask/modalAddTask.ts, line: 102 
            Property 'id' does not exist on type '{}'. 

     L101:  let handler;
     L102:  if( this.task.id ) {
     L103:    handler = this.editTask(this.task, this.project);
```

If i replace `task = {};` by `task: any;` the problem seem solve on my side. (on top of file :19 when we create some variable)

**Review by RignonNoel — CHANGES_REQUESTED** [review]
Very good job !  👍  
I just have some little bugs before accept this PR

---

When i go to edit a task, i click to assign a coworker and when i click to validate the assignement the popover to edit the task reopen on the top of the navbar (see screenshot)

screenshot from 2017-03-22 13-50-19

--- 

On the same screenshot we see the title of the edit page where we can read "New task". We need to make a condition to display "Edit task"

**Comment by RignonNoel** [comment]
Hum, i've spoke with a friend here and he propose to just put a pencil in place of the popover to edit the task because it's the main action (and why not add the option button on the right of the pencil the day we need)

I think it's not stupid. I let you see what you love ;)

File: tasklist/src/pages/modalAddTask/modalAddTask.ts (hunk 1 above)
Comment (kristofbc) [reply tasklist/src/pages/modalAddTask/modalAddTask.ts]: Oh wow, I did not see that in my debugger, I'll fix it!

**Comment by kristofbc** [comment]
@RignonNoel There you go! Indeed, the pencil is a better option. I did not removed the popover logic code, only the html button. I also corrected the title and the JS error!

**Review by RignonNoel — CHANGES_REQUESTED** [review]
Yeah seems good!

I've just two other little problem now but i think it's the same one problem with the bidding of model.

---
1. I go on the detail of a task
2. I click on edit
3. I change the priority
4. I click on the cross to cancel edition
5. The task seems edit (bug)
6. I pop to the list task
7. I select the same task to refresh the detail
8. The task have the correct content (not change)
---
1. I go on the detail of a page
2. I click on the edit button
3. I click on the select field to assign people
4. I select a people allowed in the list
5. I accept to close the modal
6. The people is not assign

 - If i repeat this the 4,5,6 just after it works

---

But very nice work @kristofbc ! ;)

**Comment by kristofbc** [comment]
@RignonNoel Indeed, there was a problem with the model binding! Nice spotting that! It should be good now!

**Review by RignonNoel — CHANGES_REQUESTED** [review]
Yeah! Now it works good with the sync ;)
But the assigned people doesn't work again, in fact, i can't assign new people now. 

1. I go on the detail of a page
2. I click on the edit button
3. I click on the select field to assign people
4. I select a people allowed in the list
5. I accept to close the modal
6. The people is not assign

If i repeat 4,5,6 just after it work

When i save the edit page, the people is not assigned in all case. Strange..

**Comment by kristofbc** [comment]
@RignonNoel The problem was behind the data binding, indeed. When we first create a task, we don't know what is a `assigned` properties composed of inside a `Task` (which is a `User`). But when we're editing a task, we know that it's an object containing the properties of a `User`. Because the database return all the properties of a `Task`. 
I thought it was quite confusing, thus, I've completed the data model: instead of using plain ol' object, we now use models that represent the data, across the application. Which is far more readable and maintainable. What do you think?
This change also fix the problem with the task assignation.

**Review by RignonNoel — APPROVED** [review]
I think it's very good move! It works perfectly! And the new architecture of model it's a good enhancement to the future! 👍 

I'm just not a big fan of the `props()` name beacause i love full name standard `getProperties()`.

**Comment by RignonNoel** [comment]
I let you merge your PR if you're ok ;)

**Comment by kristofbc** [comment]
@RignonNoel I've renames props() to getProperties() and merged the branch

**Pull request merged**